File: auth-adapter-in/src/main/java/com/example/auth/adapter/in/rest/SessionResponse.java
package com.example.auth.adapter.in.rest;

import com.example.auth.application.port.in.ListMySessionsUseCase.SessionView;
import java.time.Instant;
import java.util.UUID;

public record SessionResponse(
        UUID sessionId,
        String deviceLabel,
        String ipAddress,
        Instant issuedAt,
        Instant lastUsedAt,
        Instant expiresAt) {

    public static SessionResponse from(SessionView v) {
        return new SessionResponse(
                v.sessionId(), v.deviceLabel(), v.ipAddress(),
                v.issuedAt(), v.lastUsedAt(), v.expiresAt());
    }
}

File: auth-adapter-in/src/main/kotlin/com/example/auth/adapter/in/rest/SessionResponse.kt
package com.example.auth.adapter.`in`.rest

import com.example.auth.application.port.`in`.ListMySessionsUseCase.SessionView
import java.time.Instant
import java.util.UUID

@JvmRecord
data class SessionResponse(
    val sessionId: UUID,
    val deviceLabel: String?,
    val ipAddress: String?,
    val issuedAt: Instant,
    val lastUsedAt: Instant?,
    val expiresAt: Instant,
) {

    companion object {

        @JvmStatic
        fun from(v: SessionView): SessionResponse = SessionResponse(
            v.sessionId,
            v.deviceLabel,
            v.ipAddress,
            v.issuedAt,
            v.lastUsedAt,
            v.expiresAt,
        )
    }
}
